diff --git i/include/import/eb32tree.h w/include/import/eb32tree.h
index 1c03fc1ed..daa9e8269 100644
--- i/include/import/eb32tree.h
+++ w/include/import/eb32tree.h
@@ -301,9 +301,6 @@ __eb32_insert(struct eb_root *root, struct eb32_node *new) {
 	 * would sit on different branches).
 	 */
 
-	// note that if EB_NODE_BITS > 1, we should check that it's still >= 0
-	new->node.bit = flsnz(new->key ^ old->key) - EB_NODE_BITS;
-
 	if (new->key == old->key) {
 		new->node.bit = -1; /* mark as new dup tree, just in case */
 
@@ -321,7 +318,10 @@ __eb32_insert(struct eb_root *root, struct eb32_node *new) {
 			return container_of(ret, struct eb32_node, node);
 		}
 		/* otherwise fall through */
-	}
+	} else {
+ 	    // note that if EB_NODE_BITS > 1, we should check that it's still >= 0
+	    new->node.bit = flsnz(new->key ^ old->key) - EB_NODE_BITS;
+    }
 
 	if (new->key >= old->key) {
 		new->node.branches.b[EB_LEFT] = troot;
