diff --git a/include/tscore/Diags.h b/include/tscore/Diags.h
index ffb44efda..7918cf650 100644
--- a/include/tscore/Diags.h
+++ b/include/tscore/Diags.h
@@ -188,7 +188,7 @@ diags()
 //
 #define Debug(TAG, ...)                   \
   do {                                    \
-    if (unlikely(diags()->on())) {        \
+    if (unlikely(diags() && diags()->on())) {        \
       static DbgCtl Debug_ctl(TAG);       \
       if (Debug_ctl.tag_on()) {           \
         DbgPrint(Debug_ctl, __VA_ARGS__); \
